dowhy/causal_estimator.py

```diff
@@ -679,6 +685,91 @@ def signif_results_tostr(self, signif_results):
         return s
 
 
+def estimate_effect(
+    treatment: Union[str, List[str]],
+    outcome: Union[str, List[str]],
+    identified_estimand: IdentifiedEstimand,
+    identifier_name: str,
+    method: CausalEstimator,
+    control_value: int = 0,
+    treatment_value: int = 1,
+    test_significance: Optional[bool] = None,
```

Comment:
what should be desired functionality for these parameters? test_significance, evaluate_effect_strength, confidence_intervals, effect_modifiers, etc.?

Ideally, these should be passed to the estimator. It makes sense if the user does not provide any `method`, so then we can add these parameters when initializing the Estimator class. 

But if user has provided the estimator object, then it is tricky. One option is to make these fit-related params as optional params to the estimate_effect method of an estimator. The other is to simply update them as we do with `update_input` method. 

My thinking is that test_signifinance, evaluate_effect_strength confidence_intervals do not belong in the constructor for an estimator. This would involve a refactor of all estimator classes, so it is a big step. What do you think?